components/txn_types/src/timestamp.rs

```diff
@@ -144,12 +145,14 @@ impl TsSet {
 
     pub fn from_u64s(ts: Vec<u64>) -> Self {
         // This conversion is safe because TimeStamp is a transparent wrapper over u64.
+        #[allow(clippy::unsound_collection_transmute)]
```

Comment:
This is undefined behavior. Better to fix this through iteration like `ts.into_iter().map(TimeStamp::from).collect()`

Reply:
I think it will be 100% safe if we add `#[repr(transparent)]` to `TimeStamp`.

Reply:
But `TimeStamp` is just a wrapper of u64. Seems these warnings are only due to clippy's check is more strict

Reply:
@sticnarf Tried it before, `repr(C)` eliminates `transmute_undefined_repr`, whereas `repr(transparent)` eliminates `unsound_collection_transmute`. There is always a clippy error anyway. 🤣

Reply:
Though `repr(transparent)` is more appropriate, will change it from `repr(C)`